### DISCUSSION TITLE
Vanilla Knockback Help

### DISCUSSION THREAD

**Discussion by thesamgordon** [Minestom/Minestom#1422] [Help] [answered]
Hi,

I'm trying to replicate Vanilla knockback and I'm using the Player#takeKnockBack method, here is my code.

`                target.takeKnockback((float) 0.4,
                        Math.sin(target.getPosition().yaw() * (Math.PI/180)),
                        -(Math.cos(target.getPosition().yaw() * (Math.PI / 180))));
`

The javadoc says that this is the way to replicate it, but the player being damaged is only being launched forward in one direction regardless of where I hit them. Full code can be found here (EntityAttackEvent): https://pastebin.com/q22bsq7u

**Comment by Arektor** [comment, marked as answer]
You need to retrieve the yaw of the source entity, not the target.
`target.takeKnockback((float) 0.4, Math.sin(player.getPosition().yaw() * (Math.PI/180)), -(Math.cos(player.getPosition().yaw() * (Math.PI / 180))));`

**Reply by thesamgordon** [reply]
Oh. Of course. Thanks :)
